sql/sql_optimizer.cc

```diff
@@ -10157,8 +10163,18 @@ static bool internal_remove_eq_conds(THD *thd, Item *cond,
       if (internal_remove_eq_conds(thd, item, &new_item, &tmp_cond_value, part))
         return true;
 
-      if (new_item == NULL)
-        li.remove();
+			if (new_item == NULL)
```

Comment:
using nullptr

Reply:
If the original code is modified, it will break the whole verification rule, right?